spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerFactoryProcessor.java

```diff
@@ -121,6 +61,75 @@ public int getOrder() {
   @Override
   public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
       throws BeansException {
-    this.beanDefinitionRegistry = registry;
+    if (registry instanceof BeanFactory) {
```

Comment:
Is there a way to prevent from casting?

Reply:
No

Reply:
I suppose we could make this whole thing an `ApplicationContextInitializer` instead of a `BeanDefinitionRegistryPostProcessor`. But they are equivalent and both the downcast and the initializer only but always work with Spring Boot.

Reply:
How about to revert the if condition then, so we have flat (and less indented) code?
Also do the casting once and re-use it from within the loop.

Reply:
I wouldn't want to do that, and risk a class cast exception, even theoretically. If indented code hurts your eyes you can propose a patch later, I guess.

Not sure what you mean about the re-use. It is only used once, so it's neater to do it in line IMO.

Reply:
> How about to revert the if condition then, so we have flat (and less indented) code?

Excuse my language! `Revert if condition` means, not remove it but:

if (!(registry instanceof BeanFactory)) {
  return;
}
// Current code goes here with no indentation

Reply:
> Not sure what you mean about the re-use. It is only used once, so it's neater to do it in line IMO.

```
BeanFactory bf = (BeanFactory) registry;
for (String name : registry.getBeanDefinitionNames()) {
        Class<?> cls = bf.getType(name);
```
Do not do the cast on every bean name from within the loop, but do it once prior the loop.

Reply:
That's an extra line of code. I don't see the value.

Reply:
Your take is final!